### ISSUE TITLE
Browserslist: caniuse-lite is outdated.

### ISSUE THREAD

**Issue by zebrassimo** [victronenergy/venus-docker#16]
Hi all!

tried to use the ./build.sh running on Mac M1 using Docker version 4.5.0 (74594). All runs well until this happens:

 > [html5-app 6/6] RUN npm run build:
#43 0.473 
#43 0.473 > victron-venus-metrics@2.2.1 build
#43 0.473 > GENERATE_SOURCEMAP=false node scripts/build.js
#43 0.473 
#43 1.746 Creating an optimized production build...
#43 2.156 Error: error:0308010C:digital envelope routines::unsupported
#43 2.156     at new Hash (node:internal/crypto/hash:71:19)
#43 2.156     at Object.createHash (node:crypto:133:10)
#43 2.156     at module.exports (/node_modules/webpack/lib/util/createHash.js:135:53)
#43 2.156     at NormalModule._initBuildHash (/node_modules/webpack/lib/NormalModule.js:417:16)
#43 2.156     at handleParseError (/node_modules/webpack/lib/NormalModule.js:471:10)
#43 2.156     at /node_modules/webpack/lib/NormalModule.js:503:5
#43 2.156     at /node_modules/webpack/lib/NormalModule.js:358:12
#43 2.156     at /node_modules/loader-runner/lib/LoaderRunner.js:373:3
#43 2.156     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
#43 2.156     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
#43 2.156     at /node_modules/loader-runner/lib/LoaderRunner.js:236:3
#43 2.156     at runSyncOrAsync (/node_modules/loader-runner/lib/LoaderRunner.js:130:11)
#43 2.156     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
#43 2.156     at Array.<anonymous> (/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
#43 2.156     at Storage.finished (/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
#43 2.156     at /node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
#43 2.161 Browserslist: caniuse-lite is outdated. Please run:
#43 2.161   npx browserslist@latest --update-db
#43 2.161   Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
#43 2.175 /scripts/build.js:9
#43 2.175   throw err
#43 2.175   ^
#43 2.175 
#43 2.175 Error: error:0308010C:digital envelope routines::unsupported
#43 2.175     at new Hash (node:internal/crypto/hash:71:19)
#43 2.175     at Object.createHash (node:crypto:133:10)
#43 2.175     at module.exports (/node_modules/webpack/lib/util/createHash.js:135:53)
#43 2.175     at NormalModule._initBuildHash (/node_modules/webpack/lib/NormalModule.js:417:16)
#43 2.175     at /node_modules/webpack/lib/NormalModule.js:452:10
#43 2.175     at /node_modules/webpack/lib/NormalModule.js:323:13
#43 2.175     at /node_modules/loader-runner/lib/LoaderRunner.js:367:11
#43 2.175     at /node_modules/loader-runner/lib/LoaderRunner.js:233:18
#43 2.175     at context.callback (/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
#43 2.175     at /node_modules/babel-loader/lib/index.js:59:103
#43 2.175     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
#43 2.175   opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
#43 2.175   library: 'digital envelope routines',
#43 2.175   reason: 'unsupported',
#43 2.175   code: 'ERR_OSSL_EVP_UNSUPPORTED'
#43 2.175 }
#43 2.175 
#43 2.175 Node.js v18.12.1`

**Comment by zebrassimo** [comment]
Updating to Docker Desktop 4.15.0 (93002) didn't help.

Altering the dockerfile as below
```
FROM node:lts-alpine as html5-app
COPY venus-html5-app/package.json .
COPY venus-html5-app/package-lock.json .
RUN npm install
COPY venus-html5-app .
ENV PUBLIC_URL=/
ENV REACT_APP_ENABLE_LANG_OVERRIDE=true
RUN npx browserslist@latest --update-db
RUN npm run build
```

would still produce the error
```
#46 1.420 Creating an optimized production build...
#46 1.705 Error: error:0308010C:digital envelope routines::unsupported
#46 1.705     at new Hash (node:internal/crypto/hash:71:19)
#46 1.705     at Object.createHash (node:crypto:133:10)
#46 1.705     at module.exports (/node_modules/webpack/lib/util/createHash.js:135:53)
#46 1.705     at NormalModule._initBuildHash (/node_modules/webpack/lib/NormalModule.js:417:16)
#46 1.705     at handleParseError (/node_modules/webpack/lib/NormalModule.js:471:10)
#46 1.705     at /node_modules/webpack/lib/NormalModule.js:503:5
#46 1.705     at /node_modules/webpack/lib/NormalModule.js:358:12
#46 1.705     at /node_modules/loader-runner/lib/LoaderRunner.js:373:3
#46 1.705     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
#46 1.705     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
#46 1.705     at /node_modules/loader-runner/lib/LoaderRunner.js:236:3
#46 1.705     at runSyncOrAsync (/node_modules/loader-runner/lib/LoaderRunner.js:130:11)
#46 1.705     at iterateNormalLoaders (/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
#46 1.705     at Array.<anonymous> (/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
#46 1.705     at Storage.finished (/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
#46 1.705     at /node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
#46 1.731 /scripts/build.js:9
```
After doing some research on [stackexchange](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported), it's clear this was caused by a node.js fix. Running
`RUN npm audit fix --force`
also didn't provide a fix.

`RUN export NODE_OPTIONS=--openssl-legacy-provider && yarn build && yarn install --production --ignore-scripts --prefer-offline`
also not.

In short: error:0308010C:digital envelope routines::unsupported ws introduced as a breaking fix to a security upgrade of the node.js team which needs to be fixed in the Victron venus-docker. I wasn't able to though.

**Comment by AlexImb** [comment]
Just built a new image on an M1 Mac with the latest submodules and it worked.

**Comment by martinbosma** [comment]
Was fixed at some point, closing
